doc/source/whatsnew/v1.2.1.rst

```diff
@@ -34,6 +34,7 @@ Fixed regressions
 - Fixed regression in repr of float-like strings of an ``object`` dtype having trailing 0's truncated after the decimal (:issue:`38708`)
 - Fixed regression that raised ``AttributeError`` with PyArrow versions [0.16.0, 1.0.0) (:issue:`38801`)
 - Fixed regression in :func:`pandas.testing.assert_frame_equal` raising ``TypeError`` with ``check_like=True`` when :class:`Index` or columns have mixed dtype (:issue:`39168`)
+- Fixed regression in :meth:`~DataFrame.to_csv` opening `codecs.StreamReaderWriter` in binary mode instead of in text mode (:issue:`39247`)
```

Comment:
@jorisvandenbossche did a cleanup to re-order the regression fixes, #39246

can you move adjacent to the other IO regressions

Reply:
I moved it up to the other `to_csv` issue.

Reply:
Thanks a lot!